timely/src/dataflow/operators/core/partition.rs

```diff
@@ -0,0 +1,73 @@
+//! Partition a stream of records into multiple streams.
+
+use timely_container::{Container, PushInto, SizableContainer};
+
+use crate::dataflow::channels::pact::Pipeline;
+use crate::dataflow::operators::generic::builder_rc::OperatorBuilder;
+use crate::dataflow::{Scope, StreamCore};
+use crate::Data;
+
+/// Partition a stream of records into multiple streams.
+pub trait Partition<G: Scope, C: Container> {
+    /// Produces `parts` output streams, containing records produced and assigned by `route`.
+    ///
+    /// # Examples
+    /// ```
+    /// use timely::dataflow::operators::ToStream;
+    /// use timely::dataflow::operators::core::{Partition, Inspect};
+    ///
+    /// timely::example(|scope| {
+    ///     let streams = (0..10).to_stream(scope)
+    ///                          .partition(3, |x| (x % 3, x));
+    ///
+    ///     for (idx, stream) in streams.into_iter().enumerate() {
+    ///         stream
+    ///             .container::<Vec<_>>()
+    ///             .inspect(move |x| println!("seen {idx}: {x:?}"));
+    ///     }
+    /// });
+    /// ```
+    fn partition<C2, D2, F>(&self, parts: u64, route: F) -> Vec<StreamCore<G, C2>>
```

Comment:
Consider taking a container builder here instead!

Reply:
Good point! Changed to a container builder argument